client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -73,7 +81,7 @@ export default React.createClass( {
 		const sanitizedFieldValues = Object.assign( {}, fieldValues );
 		this.fieldNames.forEach( ( fieldName ) => {
 			if ( typeof fieldValues[ fieldName ] === 'string' ) {
-				sanitizedFieldValues[ fieldName ] = fieldValues[ fieldName ].trim();
+				sanitizedFieldValues[ fieldName ] = deburr( fieldValues[ fieldName ].trim() );
```

Comment:
> I'd rather not - the sanitizerFunction feature of formState.Controller is there for a reason

... I'm not sure of that 😂 I think updating the input will provide a degraded user experience and we have been doing a bit of custom handling at `handleChangeEvent` anyway. 

We have a delay in sanitize function is because it's debounced. FormStateController allows to define a custom debounce timer, but that uses the same value for both sanitize and validate, so it's a no go.

I agree that it makes things a little bit more complicated, but when comparing the added complexity vs the UX, I think UX wins here.

I'll probably be okay if you insist on keeping this as is, though.